### DISCUSSION TITLE
How to connect using Unix Sockets?

### DISCUSSION THREAD

**Discussion by Rockbass** [mikro-orm/mikro-orm#4525] [Q&A]
Hi there!

Could anybody tell is it possible to use mikro-orm connecting to Unix Socket instead of TCP connection? 

A bit more details: 
I'd like to connect to Google Cloud SQL inside my App Engine Flexible application using Unix Socket instead of TCP connection. 
According to [their examples](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/cloud-sql/postgres/knex/connect-unix.js), Knex with `pg` lib, which Mikro-Orm is using internally, are absolutely able to do this. 

I'm on `5.7.12` version (seems to be latest), tried to pass single `clientString` option as well as passing socket connection as `host` param along with database name and username/password. No luck. 

It seems mikro-orm internally is [always trying to parse](https://github.com/mikro-orm/mikro-orm/blob/16fbdca12743ff43357bd431bb7970d4d7870bca/packages/core/src/connections/Connection.ts#L74) clientString as URL in order to use its parts in other parts of functionality (like logging successful connection partially): 

`MikroORM successfully connected to database development on postgresql://postgres:*****@localhost:5432`

I've tried to transform my connection string (`clientUrl`) using different approaches listed [here](https://www.postgresql.org/docs/9.2/libpq-connect.html) like passing params as query params, using `socket:` prefix instead of `postgresql` and so on. Still no luck.

I will really appreciate any advice.

**Comment by B4nan** [comment]
You can use `driverOptions` to override anything that is passed down to knex.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/knex/src/AbstractSqlConnection.ts#L153-L157
